src/modules/cinema/cinema.schema.ts

```diff
@@ -16,27 +18,33 @@ export class Cinema {
   @Prop({ type: Number })
   numberOfHalls: number;
 
-  @Prop({ required: true, type: Array })
+  @Prop({ type: Array })
   menu: CreateMenuItem[];
 
-  @Prop({ required: true, type: Array })
-  projections: CreateProjection[];
+  @Prop({ type: [{ type: mongoose.Schema.Types.ObjectId, ref: 'Projection' }] })
+  projections: Projection[];
```

Comment:
Since there are not a lot of cases when we need to get the cinema with all of its projections, it's better to keep only cinemaId for projection document and for cinema you can set up a virtual property 'projections' with ref set to 'Projection', local filed '_id' and foreign field 'cinemaId'. This way you still will be able to populate and get the cinema with all its projections, but you won't need to update the array of projection IDs manually each time you add/remove projection